src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -138,6 +138,18 @@ public sealed class CosmosFhirDataStore : IFhirDataStore, IProvideCapability
                 {
                     // this means there is already an existing version of this resource
                 }
+                catch (CosmosException e) when (e.StatusCode == HttpStatusCode.ServiceUnavailable)
+                {
+                    if ((e.Message != null && e.Message.Contains("RequestTimeout", StringComparison.OrdinalIgnoreCase)) ||
+                        (e.InnerException != null && e.InnerException.Message != null && e.InnerException.Message.Contains("RequestTimeout", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        throw new CosmosException(e.Message, HttpStatusCode.RequestTimeout, e.SubStatusCode, e.ActivityId, e.RequestCharge);
+                    }
```

Comment:
Should this be an extension like `IsRequestRateExceeded`? This is doing more checks than in the RetryExceptionPolicyFactory.cs file

Reply:
Agreed, it seems cleaner in the extensions class.